File: src/main/java/um/haberes/rest/model/Curso.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Curso extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -1540771929062166596L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cursoId;

	private String nombre = "";
	private Integer facultadId;
	private Integer geograficaId;
	private Byte anual;
	private Byte semestre1;
	private Byte semestre2;
	private Integer nivelId;
	private Byte adicionalCargaHoraria;

	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;
	
	@OneToOne
	@JoinColumn(name = "nivelId", insertable = false, updatable = false)
	private Nivel nivel;

}

File: src/main/java/um/haberes/rest/kotlin/model/Curso.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import um.haberes.rest.model.Facultad
import um.haberes.rest.model.Geografica
import um.haberes.rest.model.Nivel

@Entity
@Table
data class Curso(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cursoId: Long? = null,

    var nombre: String = "",
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var anual: Byte = 0,
    var semestre1: Byte = 0,
    var semestre2: Byte = 0,
    var nivelId: Int? = null,
    var adicionalCargaHoraria: Byte = 0,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: Facultad? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: Geografica? = null,

    @OneToOne
    @JoinColumn(name = "nivelId", insertable = false, updatable = false)
    var nivel: Nivel? = null

) : Auditable()
